Add script to sync engines with core version

mischnic opened the pull request:
Closes https://github.com/parcel-bundler/parcel/issues/6854

```
> require("semver").satisfies("2.0.0-nightly.893+da998f75", "^2.0.0")
false
> require("semver").satisfies("2.0.0-nightly.893+da998f75", "2.0.0-nightly.893")
true
> require("semver").satisfies("2.0.0-nightly.893+da998f75", "2.0.0-nightly.892")
false
```

- [x] The only question is when this should be called

mischnic committed:
Add script to sync engines with core version

mischnic committed:
Remove unused scripts

parcel-benchmark commented:
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.73s | -6.00ms |
| Cached | 267.00ms | -6.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 519.00ms | -36.00ms 🚀 |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 520.00ms | -36.00ms 🚀 |
| dist/modern/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 520.00ms | -36.00ms 🚀 |

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 9.09s | -171.00ms |
| Cached | 425.00ms | -9.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 59.49s | -208.00ms |
| Cached | 1.62s | +25.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.f0cb9522.js | 693.77kb | +0.00b | 33.73s | -11.81s 🚀 |
| dist/editorView.af442747.js | 594.92kb | +0.00b | 33.73s | -11.42s 🚀 |
| dist/popup.64bc9a82.js | 209.67kb | +0.00b | 33.73s | -11.42s 🚀 |
| dist/index.87a1d973.js | 145.62kb | +0.00b | 33.73s | -11.81s 🚀 |
| dist/Toolbar.e3ba4416.js | 107.23kb | +0.00b | 33.73s | -11.42s 🚀 |
| dist/card.8ef34b4f.js | 62.67kb | +0.00b | 33.73s | -11.81s 🚀 |
| dist/Modal.cd71eaf3.js | 45.33kb | +0.00b | 33.73s | -11.42s 🚀 |
| dist/js.660b75db.js | 17.25kb | +0.00b | 33.73s | -11.42s 🚀 |
| dist/ui.5d3f7adc.js | 14.94kb | +0.00b | 33.74s | -11.41s 🚀 |
| dist/smartMediaEditor.48c8cf63.js | 13.25kb | +0.00b | 33.73s | -11.42s 🚀 |
| dist/dropzone.e0209bf8.js | 12.15kb | +0.00b | 33.73s | -11.42s 🚀 |
| dist/card.26823fa1.js | 5.77kb | +0.00b | 33.73s | -11.81s 🚀 |
| dist/EmojiPickerComponent.0482d6c0.js | 3.73kb | +0.00b | 33.74s | -11.41s 🚀 |
| dist/png-chunks-extract.5c9c0fb9.js | 3.58kb | +0.00b | 33.73s | -11.42s 🚀 |
| dist/dropzone.55bef257.js | 3.29kb | +0.00b | 33.73s | -11.42s 🚀 |
| dist/Modal.ecafad9e.js | 3.15kb | +0.00b | 33.73s | -11.81s 🚀 |
| dist/clipboard.df70240c.js | 2.93kb | +0.00b | 33.73s | -11.42s 🚀 |
| dist/ResourcedEmojiComponent.667554b4.js | 2.12kb | +0.00b | 33.74s | -11.41s 🚀 |
| dist/browser.4e039ed7.js | 1.69kb | +0.00b | 33.73s | -11.42s 🚀 |
| dist/media-card-analytics-error-boundary.c718a9a7.js | 1.12kb | +0.00b | 33.73s | -11.42s 🚀 |
| dist/media-picker-analytics-error-boundary.8b2547e5.js | 966.00b | +0.00b | 33.73s | -11.42s 🚀 |
| dist/media-card-analytics-error-boundary.30f51dac.js | 962.00b | +0.00b | 33.73s | -11.81s 🚀 |
| dist/simpleHasher.fc0d6100.js | 643.00b | +0.00b | 33.74s | -11.41s 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/EmojiPickerComponent.07ae4435.js | 147.12kb | +0.00b | 28.22s | +9.19s ⚠️ |
| dist/esm.f1a592e5.js | 33.15kb | +0.00b | 28.22s | +9.19s ⚠️ |
| dist/DatePicker.f2994e45.js | 22.98kb | +0.00b | 28.22s | +9.19s ⚠️ |
| dist/js.660b75db.js | 17.25kb | +0.00b | 45.66s | +10.88s ⚠️ |
| dist/png-chunks-extract.5c9c0fb9.js | 3.58kb | +0.00b | 45.66s | +10.88s ⚠️ |
| dist/feedback.c7e69064.js | 1.77kb | +0.00b | 28.22s | +9.19s ⚠️ |
| dist/workerHasher.3639b1f9.js | 1.63kb | +0.00b | 28.22s | +9.19s ⚠️ |
| dist/heading6.a693146e.js | 1.51kb | +0.00b | 28.22s | +9.20s ⚠️ |
| dist/heading5.8bfafe21.js | 1.38kb | +0.00b | 28.22s | +9.19s ⚠️ |
| dist/expand.a44d10fe.js | 1.29kb | +0.00b | 28.22s | +9.19s ⚠️ |

</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 6.30s | +7.00ms |
| Cached | 354.00ms | +16.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/314179339901993538)**

mischnic commented:
Every `lerna publish` step calls `prepublishOnly`, which is also how `scripts/update-config-dependencies.js` (to update `parcelDependencies`) was already called for nightly releases. So I just added this script there as well.

Also, all `engines.parcel` are now replaced with the corresponding `@parcel/core` version. For nightly and prereleases, it's an exact range and for normal releases a caret range
